pkg/controller/v1beta1/inferenceservice/reconcilers/deployment/deployment_reconciler.go

```diff
@@ -41,26 +45,91 @@ var log = logf.Log.WithName("DeploymentReconciler")
 
 // DeploymentReconciler reconciles the raw kubernetes deployment resource
 type DeploymentReconciler struct {
-	client       kclient.Client
-	scheme       *runtime.Scheme
-	Deployment   *appsv1.Deployment
-	componentExt *v1beta1.ComponentExtensionSpec
+	client         kclient.Client
+	scheme         *runtime.Scheme
+	DeploymentList []*appsv1.Deployment
+	componentExt   *v1beta1.ComponentExtensionSpec
 }
 
 func NewDeploymentReconciler(client kclient.Client,
 	scheme *runtime.Scheme,
 	componentMeta metav1.ObjectMeta,
+	workerComponentMeta metav1.ObjectMeta,
 	componentExt *v1beta1.ComponentExtensionSpec,
-	podSpec *corev1.PodSpec) *DeploymentReconciler {
+	podSpec *corev1.PodSpec, workerPodSpec *corev1.PodSpec) *DeploymentReconciler {
 	return &DeploymentReconciler{
-		client:       client,
-		scheme:       scheme,
-		Deployment:   createRawDeployment(componentMeta, componentExt, podSpec),
-		componentExt: componentExt,
+		client:         client,
+		scheme:         scheme,
+		DeploymentList: createRawDeployment(componentMeta, workerComponentMeta, componentExt, podSpec, workerPodSpec),
+		componentExt:   componentExt,
 	}
 }
+func createRawDeployment(componentMeta metav1.ObjectMeta, workerComponentMeta metav1.ObjectMeta,
+	componentExt *v1beta1.ComponentExtensionSpec, //nolint:unparam
+	podSpec *corev1.PodSpec, workerPodSpec *corev1.PodSpec) []*appsv1.Deployment {
+	var deploymentList []*appsv1.Deployment
+	var workerNodeSize string
+	var pipelineParallelSize string
+
+	isvcName := componentMeta.GetLabels()[constants.InferenceServiceLabel]
+	multiNodeEnabled := false
+
+	if workerComponentMeta.Name != "" {
+		multiNodeEnabled = true
+		workerNodeSize = componentMeta.GetAnnotations()[constants.WorkerNodeSize]
+		if parsedValue, err := strconv.Atoi(workerNodeSize); err == nil {
+			// Set pipelineParallelSize to workerNodeSize + 1 (head)
+			pipelineParallelSize = strconv.Itoa(parsedValue + 1)
+		} else {
+			log.Error(err, "Failed to convert pipelineParallelSize to int, using the WorkerSpec.Size)")
+		}
+	}
+
+	// Use defaut value(1) if tensor-parallel-size is not set (gpu count)
+	tensorParallelSize := constants.DefaultTensorParallelSize
```

Comment:
shouldn't this be added only if there is a workPodSpec defined?

Reply:
That's correct. It was added there because the variable is needed for both default deployment and worker deployment. I will modify this section to set the variable only in the case of multi-node deployments.